pandas/tests/indexes/test_multi.py

```diff
@@ -632,6 +632,24 @@ def test_from_arrays_index_series_period(self):
 
         tm.assert_index_equal(result, result2)
 
+    def test_from_array_index_series_categorical(self):
```

Comment:
I can add a test. But `set_levels` is fine. The issue is actually with the `Categorical` constructor - this is where the information about a categorical dtype is lost.